ibmcloud/README.md

```diff
@@ -255,82 +255,112 @@ $ install cloud-api-adaptor /usr/local/bin/
+Deploying the demo workload will create a new nginx Pod and a NodePort service on your Kubernetes cluster, and a new Virtual Server instance for the peer pod will be created in your IBM Cloud VPC. The `run-nginx-demo` Terraform template will also sniff test the deployed nginx server by accessing the HTTP port of the NodePort service and test that the CPU architecture of the Kubernetes worker matches that of the peer pod instance.
+
+> **Tip:** You can run the nginx sniff test manually if you log into the Kubernetes worker node and run the command
+> ```bash
+> $ curl http://localhost:30080
+> ```
+> You can also check the CPU architecture the pod VM instance is using by running the command
+> ```bash
+> $ kubectl exec nginx -- uname -a
+> ```
+> while logged into to Kubernetes worker node. If you are using a `s390x` based image as the pod VM image, the output looks like this.
+> ```
+> Linux nginx 5.4.0-109-generic #123-Ubuntu SMP [Date] s390x GNU/Linux
+> ```
+
+> **Note:** The cloud API adaptor establishes a network tunnel between the worker and pod VM, and the network traffic to/from the pod VM is transparently transferred via the tunnel.
 
+If you execute the `terraform destroy` command for the `run-nginx-demo` Terraform template the nginx Pod, ConfigMap and NodePort Service, as well as the RuntimeClass for Kata will be deleted on the cluster.
+
+A Terraform template that checks the nginx peer pod instance has been successfully created on your IBM Cloud VPC is available in [ibmcloud/terraform/check-podvm-instance](./terraform/check-podvm-instance).
+
+Create a `terraform.tfvars` file in the [template directory](./terraform/check-podvm-instance) for this Terraform template on your `development machine`. The `terraform.tfvars` file should look like this
 
-You can also check the pod VM instance architecture by command:
-```bash
-$ kubectl exec nginx -- uname -a
 ```
-If you are using `s390x` based image as pod vm image, the output looks like this.
+ibmcloud_api_key = "<your API Key>"
+podvm_image_name = "<name of your pod VM image>"
 ```
-Linux nginx 5.4.0-109-generic #123-Ubuntu SMP [Date] s390x GNU/Linux
+
+> **Notes:**
+> - `ibmcloud_api_key` is your IBM Cloud API Key that you created at [https://cloud.ibm.com/iam/apikeys](https://cloud.ibm.com/iam/apikeys).
+> - `podvm_image_name` is the Custom Image for VPC that was built and uploaded by running the Terraform template in [ibmcloud/terraform/podvm-build](terraform/podvm). View [IBM Cloud Custom images for VPC](https://cloud.ibm.com/vpc-ext/compute/images) for your chosen region to view the name of the pod VM custom image that was built and uploaded.
+
+Execute the following commands on your `development machine` to check the nginx pod VM instance exists after deploying the nginx demo workload:
+
+```bash
+$ cd ibmcloud/terraform/check-podvm-instance
+$ terraform init
+$ terraform plan
```

Comment:
If I already executed the `terraform destroy` command for the `run-nginx-demo` Terraform template.
I will meet follow error:
```

│ Error: Missing required argument
│
│   with data.ibm_is_instance.podvm_instance,
│   on main.tf line 22, in data "ibm_is_instance" "podvm_instance":
│   22:   name = one(local.peer_pod_vms)
│
│ The argument "name" is required, but no definition was found.
╵
```
Can we refine the logic, not throw the error when there is no nginx peer pod instances

Reply:
I changed the logic to create separate Terraform resources, which print the status of the test, upon pass and fail conditions.